src/Modules/ContentBlocks/Domain/ContentBlock/ContentBlockRepository.php

```diff
@@ -0,0 +1,122 @@
+        $entityManager->persist($contentBlock);
+        $entityManager->flush();
+
+        if ($contentBlock->getStatus() !== Status::Archived) {
+            $this->updateWidget($contentBlock);
+        }
+    }
+
+    public function remove(ContentBlock $contentBlock): void
+    {
+        $entityManager = $this->getEntityManager();
+        $entityManager->remove($contentBlock);
+        $entityManager->flush();
+    }
+
+    public function removeMultiple(array $contentBlocks): void
+    {
+        $entityManager = $this->getEntityManager();
+        /** @var ContentBlock $contentBlock */
+        foreach ($contentBlocks as $contentBlock) {
+            $entityManager->remove($contentBlock);
+        }
+        $entityManager->flush();
+    }
+
+    public function getVersionsForRevisionId(int $revisionId): array
+    {
+        $contentBlock = $this->findOneby(['revisionId' => $revisionId]);
+
+        if ($contentBlock === null) {
+            return [];
+        }
+
+        return $this->findBy(['id' => $contentBlock->getId(), 'locale' => $contentBlock->getLocale()]);
+    }
+
+    public function removeByRevisionId(int $revisionId): void
+    {
+        $contentBlock = $this->findOneby(['revisionId' => $revisionId]);
+
+        if ($contentBlock === null) {
+            return;
+        }
+
+        // get all versions
+        $versions = $this->findBy(['id' => $contentBlock->getId(), 'locale' => $contentBlock->getLocale()]);
+
+        $entityManager = $this->getEntityManager();
+        foreach ($versions as $version) {
+            $entityManager->remove($version);
+        }
+
+        $entityManager->flush();
+    }
+
+    public function getNextIdForLanguage(Locale $locale): int
+    {
+        return (int) $this->getEntityManager()
+                ->createQueryBuilder()
+                ->select('MAX(cb.id) as id')
+                ->from(ContentBlock::class, 'cb')
+                ->where('cb.locale = :locale')
+                ->setParameter('locale', $locale)
+                ->getQuery()
+                ->getSingleScalarResult() + 1;
+    }
+
+    public function findForIdAndLocale(int $id, string $language): ?ContentBlock
+    {
+        return $this->createQueryBuilder('cb')
+            ->andWhere('cb.id = :id')
+            ->andWhere('cb.status = :active')
+            ->andWhere('cb.locale = :locale')
+            ->setParameter('id', $id)
+            ->setParameter('active', Status::Active)
+            ->setParameter('locale', Locale::from($language))
+            ->getQuery()
+            ->getOneOrNullResult();
+    }
+
+    private function updateWidget(ContentBlock $contentBlock): void
+    {
+        /** @var Block $block */
+        $block = $this->blockRepository->findOneBy(['id' => $contentBlock->getExtraId()]);
+
+        $block->getSettings()->add([
+            'label' => $contentBlock->getTitle(),
+            'id' => $contentBlock->getId(),
+            'language' => $contentBlock->getLocale(),
```

Comment:
you don't need so set the language here since the locale is already set on the block itself so you can remove this entry

Reply:
already removed